runtime/src/main/java/com/tns/AppConfig.java

```diff
@@ -128,28 +109,27 @@ public AppConfig(File appDir) {
     }
 
     private static Object[] makeDefaultOptions() {
-        Object[] result = new Object[KnownKeys.asArray.length];
-        int index = 0;
-        for (KnownKeys key: KnownKeys.asArray) {
-            result[index++] = key.getDefaultValue();
+        Object[] result = new Object[KnownKeys.values().length];
+        for (KnownKeys key: KnownKeys.values()) {
+            result[key.ordinal()] = key.getDefaultValue();
         }
         return result;
     }
 
     public int getGcThrottleTime() {
-        return (int)values[KnownKeys.GcThrottleTime.getIndex()];
+        return (int)values[KnownKeys.GcThrottleTime.ordinal()];
     }
 
     public int getMemoryCheckInterval() {
-        return (int)values[KnownKeys.MemoryCheckInterval.getIndex()];
+        return (int)values[KnownKeys.MemoryCheckInterval.ordinal()];
     }
 
     public double getFreeMemoryRatio() {
-        return (double)values[KnownKeys.FreeMemoryRatio.getIndex()];
+        return (double)values[KnownKeys.FreeMemoryRatio.ordinal()];
     }
 
     public String getProfilingMode() {
-        return (String)values[KnownKeys.Profiling.getIndex()];
+        return (String)values[KnownKeys.Profiling.ordinal()];
     }
-    public String getMarkingMode() { return (String)values[KnownKeys.MarkingMode.getIndex()]; }
+    public MarkingMode getMarkingMode() { return (MarkingMode)values[KnownKeys.MarkingMode.ordinal()]; }
```

Comment:
no need for cast `(MarkingMode)`

Reply:
The values is of type Object[], indexing it returns Object.